transportable-udfs-avro/src/main/java/com/linkedin/transport/avro/AvroWrapper.java

```diff
@@ -43,32 +43,69 @@ private AvroWrapper() {
 
   public static StdData createStdData(Object avroData, Schema avroSchema) {
     switch (avroSchema.getType()) {
-      case INT:
+      case INT: {
+        if (!(avroData instanceof Integer)) {
```

Comment:
Can you extract the shared code into a method? like
private validateData(Object data, Type avroType, Class... acceptableClasses) {
...
// throws exception here if data is not an instance of acceptableClasses, you can use method `isAssignableFrom` 
}